### ISSUE TITLE
Modded chunks lose their NBT data

### ISSUE THREAD

**Issue by ViRb3** [Amulet-Team/Amulet-Map-Editor#148]
**To Reproduce**
1. Load a Forge 1.16 world, ideally with custom TileEntities and custom NBT data
2. Make any change in a chunk, For an example, delete a block
3. Save the changes
4. Open the chunk in NBTExplorer

**Expected behavior**
The NBT data is intact

**Actual behavior**
All NBT data is wiped

**Screenshots**
As you can see below, the edited chunk (red) has its NBT data reset (TileEntities), and any custom entries are deleted entirely (BloodMagic). The non-edited chunks (green) are untouched.
![image](https://user-images.githubusercontent.com/2650170/101991610-dffb7080-3ca5-11eb-8178-956c69e7da48.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version: 0.6.11 and 0.8.0b3

**Comment by gentlegiantJGC** [comment]
We don't officially support modded worlds yet.
The way Amulet works is by converting all the data into a single consistent format. The correct way to support this would be to handle data from each mod but that is unrealistic. I would have thought tile entities would get carried across though.

**Comment by ViRb3** [comment]
Would it not be possible to remember all custom/non-recognized NBT entries and re-apply them on save? Amulet doesn't need to support or even process them in any way. I am currently doing this process manually using NBTExplorer, but that's obviously not very pleasant.

**Comment by gentlegiantJGC** [comment]
This is probably what we will end up doing. The only issue is if you convert to a different version it will be carried across as is which may break things.
The other option is when saving we merge the output data with the original data so it wouldn't be carried to other worlds

**Comment by gentlegiantJGC** [comment]
I think this should solve this issue.
https://github.com/Amulet-Team/Amulet-Core/pull/111

**Comment by dadodasyra** [comment]
It's not, at least not on modified chunk, when you modify a chunk every NBT data of modded blocks are lost

**Comment by Twisted-Code** [comment]
I second what @dadodasyra said. A friend used amulet (I think the latest version) to delete some lagging pipes in a modded world, and ended up with more deletion than he bargained for. All the tile entities got nuked, not just the ones he selected (I watched him do it and I know he didn't select the whole chunk, you know?). I had to manually fix some chunks by basically reversing what he did and doing it a different way (instead of deleting with amulet, I changed the pipes to amethyst blocks in the block pallette. Seems like forge figured out "oh, amethyst blocks can't have that tile entity, I'll delete it" well enough.)
While I do remember reading the FAQ item about modded worlds, and we backed up the server beforehand, I think I would've liked to see something specifically mentioning modded worlds in the disclaimer page (especially since you require a lot more capabilities from my browser to load the FAQ than you do the mostly static content on the disclaimer, so my script blocker was blocking it initially. You might see that as a "me problem", but I beg to differ)
